assets/app/scripts/directives/fromFile.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .directive("fromFile", function($q,
+                                  $uibModal,
+                                  $location,
+                                  $filter,
+                                  CachedTemplateService,
+                                  Navigate,
+                                  TaskList,
+                                  DataService,
+                                  APIService) {
+    return {
+      restrict: "E",
+      scope: false,
+      templateUrl: "views/directives/from-file.html",
+      controller: function($scope) {
+        var aceEditorSession;
+        var humanize = $filter('humanize');
+        var humanizeKind = $filter('humanizeKind');
+        TaskList.clear();
+
+        $scope.aceLoaded = function(editor) {
+          aceEditorSession = editor.getSession();
+          aceEditorSession.setOption('tabSize', 2);
+          aceEditorSession.setOption('useSoftTabs', true);
+          editor.setDragDelay = 0;
+          editor.$blockScrolling = Infinity;
+
+          $('#from-file .editor').animate({
+            height: Math.floor(window.innerHeight * 0.50) + 'px'
+          }, 30);
+        };
+
+        // Determine whats the input format (JSON/YAML) and set appropriate view mode
+        var updateEditorMode = _.debounce(function(){
+          try {
+            JSON.parse($scope.editorContent);
+            aceEditorSession.setMode("ace/mode/json");
+          } catch (e) {
+            try {
+              YAML.parse($scope.editorContent);
+              aceEditorSession.setMode("ace/mode/yaml");
+            } catch (e) {
+              return;
+            }
+          }
+        }, 500);
+
+        // Check if the editor isn't empty to disable the 'Add' button. Also check in what format the input is in (JSON/YAML) and change
+        // the editor accordingly.
+        $scope.aceChanged = updateEditorMode();
```

Comment:
What you have now will just set `aceChanged` to the return value of `updateEditorMode`, which is `undefined`.

```
$scope.aceChanged = updateEditorMode;
```